docs/operation/Recover_From_Etcd_Permanent_Quorum_Loss.md

```diff
@@ -0,0 +1,85 @@
+
+**Note:** Please note that manually restoring etcd can result in data loss. This guide is the last resort to bring an ETCD cluster up and running again.
+
+   1. Add the following annotation to the `Etcd` resource `kubectl annotate etcd etcd-main druid.gardener.cloud/ignore-reconciliation="true"`
+   3. Note down the configmap name that is attached to the `etcd-main` statefulset. If you describe the statefulset with `kubectl describe sts etcd-main`, look for the lines similar to following lines to identify attached configmap name. It will be needed at later stages:
+
+   ```
+     Volumes:
+      etcd-config-file:
+       Type:      ConfigMap (a volume populated by a ConfigMap)
+       Name:      etcd-bootstrap-4785b0
+       Optional:  false
+   ```
+
+   3. Scale down the `etcd-main` statefulset replicas to `0`
+
+      `kubectl scale sts etcd-main --replicas=0`
+   4. The PVCs will look like the following on listing them with the command `kubectl get pvc` :
+      
+      ```
+      main-etcd-etcd-main-0        Bound    pv-shoot-garden-aws-c1c6154e-6506-4017-9cfc-4f1cc9afa730   80Gi       RWO            gardener.cloud-fast   385d
+      ```
+      Delete all PVCs that are attached to `etcd-main` cluster.
+
+      `kubectl delete pvc -l instance=etcd-main`
+   5. Edit the `etcd-main` cluster's configmap (ex: `etcd-bootstrap-4785b0`) as follows:
+      
+      Find the `initial-cluster` field in the configmap. It will look like the following:
+      ```
+      # Initial cluster
+        initial-cluster: etcd-main-0=https://etcd-main-0.etcd-main-peer.default.svc:2380,etcd-main-1=https://etcd-main-1.etcd-main-peer.default.svc:2380,etcd-main-2=https://etcd-main-2.etcd-main-peer.default.svc:2380
+      ```
+
+      Change the `initial-cluster` field to have only one member (`etcd-main-0`) in the string. It should now look like this:
+
+      ```
+      # Initial cluster
+        initial-cluster: etcd-main-0=https://etcd-main-0.etcd-main-peer.default.svc:2380
+      ```
+   6. Scale up the `etcd-main` statefulset replicas to `1`
+
+      `kubectl scale sts etcd-main --replicas=1`
+   7. Wait for the single-member etcd cluster to be completely ready.
+
+      `kubectl get pods etcd-main-0` will give the following output when ready:
+      ```
+      NAME          READY   STATUS    RESTARTS   AGE
+      etcd-main-0   2/2     Running   0          1m
+      ```
+   8. Remove the following annotation from the `Etcd` resource `etcd-main`: `kubectl annotate etcd etcd-main druid.gardener.cloud/ignore-reconciliation-`
+   9. Add the following annotation to the `etcd-main` statefulset `kubectl annotate sts etcd-main gardener.cloud/scaled-to-multi-node="true"`
```

Comment:
Can you reverse these two steps?
We should add the `scaled-to-multi-node` annotation to the sts before removing the `ignore-reconciliation` annotation from the etcd resource

Reply:
Done